components/datetime/src/format/neo.rs

```diff
@@ -648,6 +655,47 @@ impl<C: CldrCalendar> TypedDateTimePatternInterpolator<C> {
             _calendar: PhantomData::default(),
         }
     }
+
+    pub fn load_for_pattern<'l, P>(
+        &'l mut self,
+        provider: &P,
+        pattern: &'l Pattern,
+    ) -> Result<DateTimePatternFormatter<'l, C>, Error>
+    where
+        P: DataProvider<C::YearSymbolsV1Marker>
+            + DataProvider<C::MonthSymbolsV1Marker>
+            + DataProvider<WeekdaySymbolsV1Marker>
+            + DataProvider<DayPeriodSymbolsV1Marker>
+            + DataProvider<WeekDataV1Marker>
+            + ?Sized,
+    {
+        let mut required_data = RequiredDataNeo::default();
+        // TODO: Consider support for time zones here
+        required_data.add_requirements_from_pattern(pattern, false);
+        if let Some(field) = required_data.year_symbols {
+            self.load_year_names(provider, field.length)?;
+        }
+        if let Some(field) = required_data.month_symbols {
+            let FieldSymbol::Month(month) = field.symbol else {
```

Comment:
nit: i don't think we should panic in release mode here.

and i'm not entirely happy about the invariant existing in the first place, but can't be helped

Reply:
Already fixed in f53809a